lib/exceptions.rb

```diff
@@ -0,0 +1,7 @@
+module ManageIQ
+  module API
+    module Common
+      class HeadersNotSet < StandardError; end
```

Comment:
Should this inherit from `ArgumentError`?  The headers are sort of like arguments to the request